Source/WTF/wtf/Vector.h

```diff
@@ -1026,6 +1026,9 @@ template<typename T, size_t inlineCapacity, typename OverflowHandler, size_t min
 Vector<T, inlineCapacity, OverflowHandler, minCapacity, Malloc>::Vector(const Vector& other)
     : Base(other.size(), other.size())
 {
+#ifdef __swift__
+    RELEASE_ASSERT_NOT_REACHED_WITH_MESSAGE("C++ copy constructors are not memory safe in Swift. Try to find a way to borrow the object instead.");
```

Comment:
I'm a little impressed that this works at all—that Swift will import an inline definition and generate code for it—but I think the C++ model relies on the possibility of multiple compilation units generating the same template function, and the linker being allowed to pick on arbitrarily.

i.e. if Swift instantiates a `Vector<P>` but some C++ code also uses a `Vector<P>`, both compilation units will generate the same constructor, but only the former will have the release assertion. And the linker makes no guarantee about which one it'll use, but there will only be _one_ for the whole binary.

Reply:
You might be able to enforce this at compile time with `std::enable_if` or a `static_assert`.

Reply:
To give you a more rigorous answer, yes, this appears to violate the [One Definition Rule](https://en.cppreference.com/w/cpp/language/definition):

> There can be more than one definition in a program of each of the following: class type, enumeration type, inline function, inline variable(since C++17), [templated entity](https://en.cppreference.com/w/cpp/language/templates#Templated_entity) (template or member of template, but not full [template specialization](https://en.cppreference.com/w/cpp/language/template_specialization)), as long as all of the following is true:
> - each definition appears in a different translation unit
> …
> - **each definition consists of the same sequence of tokens** (typically, appears in the same header file)
> …
> 
> If all these requirements are satisfied, the program behaves as if there is only one definition in the entire program. Otherwise, the program is ill-formed, no diagnostic required.